test/integration/MExtensionSystem.t.sol

```diff
@@ -0,0 +1,1548 @@
+
+        vm.warp(vm.getBlockTimestamp() + 10 days);
+
+        // Collect MEarnerManager yield
+        yields[M_EARNER_MANAGER] += mEarnerManagerPrincipal == 0
+            ? 0
+            : _calcMEarnerManagerYield(yields[M_EARNER_MANAGER], mEarnerManagerPrincipal);
+
+        // Collect MYieldToOne yield
+        yields[M_YIELD_TO_ONE] += mBalanceBefore == 0 ? 0 : mToken.balanceOf(address(mYieldToOne)) - mBalanceBefore;
+
+        // Collect MYieldFee yield
+        uint256 priorYield = yields[M_YIELD_FEE];
+
+        yields[M_YIELD_FEE] += _calcMYieldFeeYield(amount + yields[M_YIELD_FEE], _principal);
+
+        uint256 mYieldFeeYield = mYieldFee.accruedYieldOf(alice);
+
+        assertApproxEqAbs(mYieldFeeYield, yields[M_YIELD_FEE], 50, "Should have accrued yield in mYieldFee");
+
+        return (priorYield == 0 ? amount - 2 : amount, yields);
+    }
+
+    function _testYieldCapture_mYieldToOne(
+        address from,
+        uint256[] memory yields,
+        uint256 amount
+    ) public returns (uint256, uint256[] memory) {
+        vm.prank(alice);
+        if (from == address(mToken)) swapFacility.swapInM(address(mYieldToOne), amount, alice);
+        else swapFacility.swap(from, address(mYieldToOne), amount, alice);
+
+        // Prep MEarnerManager
+        uint112 mEarnerManagerPrincipal = yields[M_EARNER_MANAGER] == 0
+            ? 0
+            : _calcMEarnerManagerPrincipal(yields[M_EARNER_MANAGER]);
+
+        // Prep MYieldFee
+        uint112 mYieldFeePrincipal = yields[M_YIELD_FEE] == 0 ? 0 : _calcMYieldFeePrincipal(yields[M_YIELD_FEE]);
+
+        // Prep MYieldToOne
+        uint256 mBalanceBefore = mToken.balanceOf(address(mYieldToOne));
+
+        vm.warp(vm.getBlockTimestamp() + 10 days);
+
+        // Collect MEarnerManager yield
+        yields[M_EARNER_MANAGER] += mEarnerManagerPrincipal == 0
+            ? 0
+            : _calcMEarnerManagerYield(yields[M_EARNER_MANAGER], mEarnerManagerPrincipal);
+
+        // Collect MYieldFee yield
+        yields[M_YIELD_FEE] += mYieldFeePrincipal == 0
+            ? 0
+            : _calcMYieldFeeYield(yields[M_YIELD_FEE], mYieldFeePrincipal);
+
+        // Assert MYieldToOne yield
+        uint256 mBalanceAfter = mToken.balanceOf(address(mYieldToOne));
+
+        uint256 mYieldToOneYield = mYieldToOne.yield();
+
+        uint256 priorYield = yields[0];
+
+        uint256 yield = mBalanceAfter - mBalanceBefore;
+
+        assertApproxEqAbs(mYieldToOneYield, yield + priorYield, 50, "Should have accrued yield in mYieldToOne");
+
+        yields[0] += yield;
+
+        return (priorYield == 0 ? amount - 2 : amount, yields);
+    }
+
+    function _testYieldCapture_mEarnerManager(
+        address from,
+        uint256[] memory yields,
+        uint256 amount
+    ) public returns (uint256, uint256[] memory) {
+        vm.prank(alice);
+        swapFacility.swap(from, address(mEarnerManager), amount, alice);
```

Comment:
No distinction between M token and other extension tokens?

Reply:
So far swapping in M is only used to bootstrap the extension loop before fuzzing permutation starts, but I can add that here, maybe the template these represent could be used differently if we get around to a new fuzz test

Reply:
Is it needed for the other functions?
https://github.com/m0-foundation/evm-m-extensions/blob/9e03e6b03de3b40de8dc3e0d8359d89d91ca2dea/test/integration/MExtensionSystem.t.sol#L1459